src/main/scala/com/fulcrumgenomics/umi/DuplexConsensusCaller.scala

```diff
@@ -140,84 +151,99 @@ class DuplexConsensusCaller(override val readNamePrefix: String,
     // Group the reads by /A vs. /B and ensure that /A is the first group and /B the second
     val groups = pairs.groupBy(r => r[String](ConsensusTags.MolecularId)).toSeq.sortBy { case (mi, _) => mi }.map(_._2)
 
-    groups match {
-      case Seq() =>
-        Nil
-      case Seq(ab) =>
-        rejectRecords(ab, FilterAorBOnly)
-        Nil
-      case Seq(ab, ba) =>
-        // Fragments have no place in duplex land (and are filtered out previously anyway)!
-        val (_, abR1s, abR2s) = subGroupRecords(ab)
-        val (_, baR1s, baR2s) = subGroupRecords(ba)
-
-        // Get all the alignments to one end of the source molecule
-        val singleStrand1 = abR1s ++ baR2s
-        val singleStrand2 = abR2s ++ baR1s
-
-        // The orientation of AB and BA reads should be:
-        // AB R1: +  AB R2: -
-        // BA R1: -  BA R2: +
-        // or vice versa (AB-R1:-, AB-R2:+, AB-R1:-, AB-R2: +
-        // Therefore, AB-R1s and BA-R2s should be on the same strand, and the same for AB-R2s and BA-R1s
-        // Check for this explicitly here.
-        if (singleStrand1.nonEmpty) {
-          val ss1Flag = singleStrand1.head.negativeStrand
-          if (singleStrand1.exists(_.negativeStrand != ss1Flag)) {
-            val ss1MI   = singleStrand1.head.apply[String](ConsensusTags.MolecularId)
-            rejectRecords(ab ++ ba, FilterCollision)
-            logger.debug(s"Not all AB-R1s and BA-R2s were on the same strand for molecule with id: $ss1MI")
-            return Nil
-          }
-        }
-        if (singleStrand2.nonEmpty) {
-          val ss2Flag = singleStrand2.head.negativeStrand
-          if (singleStrand2.exists(_.negativeStrand != ss2Flag)) {
-            val ss2MI   = singleStrand2.head.apply[String](ConsensusTags.MolecularId)
-            rejectRecords(ab ++ ba, FilterCollision)
-            logger.debug(s"Not all AB-R2s and BA-R1s were on the same strand for molecule with id: $ss2MI")
-            return Nil
-          }
-        }
+    if (!hasMinimumNumberOfReads(groups)) {
```

Comment:
I think this little block would be clearer like:

```scala
require(groups.length <= 2, "SamRecords supplied with more than two distinct MI values.")
val x = groups.head
val y = groups.lift(1).getOrElse(Seq.empty)

if (hasMinimumNumberOfReads(x, y)) {
  callDuplexConsensusRead(x, y)
else {
  rejectRecords(groups.flatten, FilterMinReads)
  Nil
}
```

I was worried that there wasn't going to be a check of <= 2 groups, and had to go hunting to find it buried in the `hasMinimumNumberOfReads()` function.